core/src/mast/mod.rs

```diff
@@ -554,6 +556,18 @@ impl fmt::Display for DecoratorId {
     }
 }
 
+impl Serializable for DecoratorId {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.0.write_into(target)
+    }
+}
+
+impl Deserializable for DecoratorId {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        Ok(Self(source.read()?))
+    }
+}
```

Comment:
Oops, forgot about that. I ended up implementing `Serializable` only (since we always want to serialize the same way, and it allows using `DecoratorList::write_into()` as a result). I changed deserialization to use `from_u32_safe()`